src/core/libraries/save_data/dialog/savedatadialog_ui.cpp

```diff
@@ -0,0 +1,787 @@
+
+#include <fmt/chrono.h>
+#include <imgui.h>
+#include <magic_enum.hpp>
+
+#include "common/singleton.h"
+#include "core/file_sys/fs.h"
+#include "core/libraries/save_data/save_instance.h"
+#include "imgui/imgui_std.h"
+#include "savedatadialog_ui.h"
+
+using namespace ImGui;
+using namespace Libraries::CommonDialog;
+
+constexpr u32 OrbisSaveDataBlockSize = 32768; // 32 KiB
+
+constexpr auto SAVE_ICON_SIZE = ImVec2{152.0f, 85.0f};
+constexpr auto SAVE_ICON_PADDING = ImVec2{8.0f, 2.0f};
+
+static constexpr ImVec2 BUTTON_SIZE{100.0f, 30.0f};
+constexpr auto FOOTER_HEIGHT = BUTTON_SIZE.y + 15.0f;
+static constexpr float PROGRESS_BAR_WIDTH{0.8f};
+
+static ::Core::FileSys::MntPoints* g_mnt =
+    Common::Singleton<::Core::FileSys::MntPoints>::Instance();
+
+static std::string SpaceSizeToString(size_t size) {
+    std::string size_str;
+    if (size > 1024 * 1024 * 1024) { // > 1GB
+        size_str = fmt::format("{:.2f} GB", double(size / 1024 / 1024) / 1024.0f);
+    } else if (size > 1024 * 1024) { // > 1MB
+        size_str = fmt::format("{:.2f} MB", double(size / 1024) / 1024.0f);
+    } else if (size > 1024) { // > 1KB
+        size_str = fmt::format("{:.2f} KB", double(size) / 1024.0f);
+    } else {
+        size_str = fmt::format("{} B", size);
+    }
+    return size_str;
+}
+
+namespace Libraries::SaveData::Dialog {
+
+void SaveDialogResult::CopyTo(OrbisSaveDataDialogResult& result) const {
+    result.mode = this->mode;
+    result.result = this->result;
+    result.buttonId = this->button_id;
+    if (result.dirName != nullptr) {
+        result.dirName->data.FromString(this->dir_name);
+    }
+    if (result.param != nullptr && this->param.GetString(SaveParams::MAINTITLE).has_value()) {
+        result.param->FromSFO(this->param);
+    }
+    result.userData = this->user_data;
+}
+
+SaveDialogState::SaveDialogState(const OrbisSaveDataDialogParam& param) {
+    this->mode = param.mode;
+    this->type = param.dispType;
+    this->user_data = param.userData;
+    if (param.optionParam != nullptr) {
+        this->enable_back = {param.optionParam->back == OptionBack::ENABLE};
+    }
+
+    static std::string game_serial{*Common::Singleton<PSF>::Instance()->GetString("CONTENT_ID"), 7,
+                                   9};
+
+    const auto item = param.items;
+    this->user_id = item->userId;
+    this->title_id = item->titleId->data;
+    if (item->titleId == nullptr) {
+        this->title_id = game_serial;
+    } else {
+        this->title_id = item->titleId->data.to_string();
+    }
+
+    for (u32 i = 0; i < item->dirNameNum; i++) {
+        const auto dir_name = item->dirName[i].data.to_view();
+
+        if (dir_name.empty()) {
+            continue;
+        }
+
+        auto dir_path = SaveInstance::MakeDirSavePath(user_id, title_id, dir_name);
+
+        PSF param_sfo;
+        param_sfo.Open(dir_path / "sce_sys" / "param.sfo");
+
+        auto utc_time = std::chrono::file_clock::to_utc(param_sfo.GetLastWrite());
```

Comment:
I'm getting a compiler error on linux, gcc 14.2.1
I can compile by changing to_utc to to_sys

This https://en.cppreference.com/w/cpp/chrono/file_clock
says
> file_clock provides exactly one of the following two pairs of static member functions:
>     to_utc and from_utc; or
>     to_sys and from_sys. 

seems weird

Reply:
Funny one. In Msvc there is no to_sys 😆

Reply:
> I'm getting a compiler error on linux, gcc 14.2.1 I can compile by changing to_utc to to_sys
> 
> This https://en.cppreference.com/w/cpp/chrono/file_clock says
> 
> > file_clock provides exactly one of the following two pairs of static member functions:
> > to_utc and from_utc; or
> > to_sys and from_sys.
> 
> seems weird

I didn't know where to post this. But I got same error on Windows while building via MSYS MINGW64.
![image](https://github.com/user-attachments/assets/a7218af4-e7c4-4c30-8e09-8425d84ce457)
So I guess wrapping it into #ifdef _WIN32 isn't enough?

Reply:
Yeah, it would need to check if it is MinGW too
`#if defined(_WIN32) && !defined(__GNUC__) && !defined(__MINGW32__) && !defined(__MINGW64__)`

Reply:
Should I create an issue?

Reply:
Can you check if this change solves your issue? If you don't want to PR, I can publish in my next PR

Reply:
![image](https://github.com/user-attachments/assets/85e24803-7d1f-4118-a19e-b716ddb8f72f)
Yeah, that one passed, now it can't find "visibility.h". cmake config issues? Should I just send it to Discord to investigate further? Cause it seems like it might be a chain of fixes

Reply:
That one's a separate issue described here: https://github.com/shadps4-emu/shadPS4/issues/1195